### PR TITLE
UIDEXP-21: Add file csv extension validation

### PR DESCRIPTION
**Pull request by Yuliia-Dovhal**
## Purpose:
Validating file with UUIDs and display an error message if invalid

## Task:
[UIDEXP-21](https://issues.folio.org/browse/UIDEXP-21)

## Screenshot:
Screenshot 2020-02-28 at 14 51 07

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,3 +6,4 @@
 * Add dependency on @folio/stripes-data-transfer-components. Add file uploader feature on landing page. Refs UIDEXP-11.
 * Add running jobs components to display temp static data. Refs UIDEXP-6.
 * Implement logic for initiating the export job process once the user has selected file. Refs UIDEXP-20.
+* Add file .csv extension validation. UIDEXP-21.
```
Comment (VictorSoroka1) [inline CHANGELOG.md]: Please make the message more clear by explaining validation for what and where?

File: src/components/QueryFileUploader/QueryFileUploader.js [hunk 2]
```diff
@@ -132,16 +153,33 @@ function QueryFileUploaderComponent(props) {
...
+          <>
+            <Layout className="padding-top-gutter padding-start-gutter padding-end-gutter textCentered">
+              <span data-test-sub-title>
+                <FormattedMessage id="ui-data-export.uploaderSubTitle" />
+              </span>
+            </Layout>
+            <ConfirmationModal
+              id="file-extension-modal"
+              open={fileExtensionModalOpen}
+              heading={<span data-test-file-extension-modal-header><FormattedMessage id="ui-data-export.modal.fileExtensions.blocked.header" /></span>}
```
Comment (YevheniiMaltsev) [inline src/components/QueryFileUploader/QueryFileUploader.js]: It seems this line is too long. Could you please split it?

File: test/bigtest/tests/query-file-uploader-test.js [hunk 3]
```diff
@@ -111,4 +114,48 @@ describe('queryFileUploader component', () => {
...
+      beforeEach(async () => {
+        await fileExtensionModal.cancelButton.click();
+      });
+
+      it('should close modal', () => {
+        expect(fileExtensionModal.isPresent).to.be.false;
+      });
+    });
+
+    describe('"Choose other files" button clicked', () => {
```
Comment (VictorSoroka1) [inline test/bigtest/tests/query-file-uploader-test.js]: Let's not hardcode the naming of things as they can be changed in the future and tests might not be updated.

File: src/components/QueryFileUploader/utils/checkFileHaveCsvExtension.js [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import { getFileExtension } from '@folio/stripes-data-transfer-components';
+
+export const checkFileHaveCsvExtension = file => {
```
Comment (YevheniiMaltsev) [inline src/components/QueryFileUploader/utils/checkFileHaveCsvExtension.js]: This case is covered by the other tests. But please, could you add a separated unit-test for this util function?

File: src/components/QueryFileUploader/QueryFileUploader.js [hunk 5]
```diff
@@ -50,18 +55,26 @@ function QueryFileUploaderComponent(props) {
...
+  };
 
-  function setDropZone(active = true) {
+  const setDropZone = (active = true) => {
     setDropZoneActive(active);
     setLoading(active);
-  }
+  };
+
+  const showFileExtensionsModal = () => {
```
Comment (VictorSoroka1) [inline src/components/QueryFileUploader/QueryFileUploader.js]:
```suggestion
  const showFileExtensionModal = () => {
```

File: src/components/QueryFileUploader/QueryFileUploader.js [hunk 6]
```diff
@@ -50,18 +55,26 @@ function QueryFileUploaderComponent(props) {
...
+    setFileExtensionModalOpen(true);
+  };
+
+  const hideFileExtensionsModal = () => {
```
Comment (VictorSoroka1) [inline src/components/QueryFileUploader/QueryFileUploader.js]:
```suggestion
  const hideFileExtensionModal = () => {
```

File: test/bigtest/interactors/queryFileUploaderInteractor.js [hunk 7]
```diff
@@ -24,4 +25,9 @@ import CalloutInteractor from '@folio/stripes-components/lib/Callout/tests/inter
   callout = new CalloutInteractor();
 }
 
+class FileExtensionsModal extends ConfirmationModalInteractor {
+  header = new Interactor('[data-test-file-extension-modal-header]');
+}
+
+export const fileExtensionModal = new FileExtensionsModal('#file-extension-modal');
```
Comment (YevheniiMaltsev) [inline test/bigtest/interactors/queryFileUploaderInteractor.js]: As far as I understand fileExtensionModal can be added to QueryFileUploaderInteractor or FileExtensionsModal should be moved to the separated file. @Yuliia-Dovhal @VictorSoroka1 what do you think about it?

Comment (VictorSoroka1) [reply test/bigtest/interactors/queryFileUploaderInteractor.js]: Sound good to add as part of `QueryFileUploaderInteractor`.

**Pull request merged**